c++/src/kj/async-io.h

```diff
@@ -59,15 +59,9 @@ class AsyncInputStream: private AsyncObject {
   virtual Promise<size_t> read(void* buffer, size_t minBytes, size_t maxBytes);
   virtual Promise<size_t> tryRead(void* buffer, size_t minBytes, size_t maxBytes) = 0;
 
-  Promise<size_t> read(ArrayPtr<byte> buffer, size_t minBytes) { 
-    return read(buffer.begin(), minBytes, buffer.size()); 
-  }
-  Promise<size_t> tryRead(ArrayPtr<byte> buffer, size_t minBytes) {
-    return tryRead(buffer.begin(), minBytes, buffer.size()); 
-  }
-  Promise<void> read(ArrayPtr<byte> buffer) {
-    return read(buffer, buffer.size()).ignoreResult();
-  }
+  Promise<size_t> read(ArrayPtr<byte> buffer, size_t minBytes);
```

Comment:
Why de-inline these? The inner code of these methods is about the same size as the function call itself so this seems like a strict loss, adding a layer of function call overhead for no improvement in code size in release mode.

Reply:
I was concerned about the ignoreResult() call, but since this function and class is non-templated it's acceptable to have it in a header. Didn't notice much difference in code size either way, so I dropped this part in the new push.

Reply:
Oh sorry, you're right that the third function here, which has `.ignoreResult()`, is significantly more complicated than a simple function call, and so might not be appropriate to inline. But this function is rarely used anyway.

The other two functions, though, are just trivial forwarding functions so really should stay inline.

Anyway, let's just leave them alone (as you have now).